src/components/MultipleChoice/BaseMultipleChoice.tsx

```diff
@@ -75,25 +79,33 @@ export const BaseMultipleChoice: FC<BaseMultipleChoiceProps> = ({
 			}
 		}
 
+		const onKeyUp = (e: KeyboardEvent) => {
+			delete keysPressedMap.current[e.key]
+		}
+
 		const eventTargetRef = getEventTarget && getEventTarget()
 
 		// Attach event listener to the event target if one is provided, otherwise attach it to the window
 		if (eventTargetRef && eventTargetRef.current) {
 			const target = eventTargetRef.current
 
 			target.addEventListener('keydown', onKeyDown)
+			target.addEventListener('keyup', onKeyUp)
 
 			return () => {
 				target.removeEventListener('keydown', onKeyDown)
+				target.removeEventListener('keyup', onKeyUp)
 			}
 		} else {
 			window.addEventListener('keydown', onKeyDown)
+			window.addEventListener('keyup', onKeyUp)
 
 			return () => {
 				window.removeEventListener('keydown', onKeyDown)
+				window.removeEventListener('keyup', onKeyUp)
 			}
 		}
-	}, [currentFocus, getEventTarget, items, onSelectedChange])
+	}, [currentFocus, getEventTarget, items, keysPressedMap, onSelectedChange])
```

Comment:
I don't think refs need to be included in the dependency array because they don't trigger rerenders